### PR TITLE
Add Example Directory And Collaboration Example

### PR DESCRIPTION
**Pull request by jcudit**
This was useful for reproducing a bug.  We can add missing examples as bugs are fixed.

/cc https://github.com/terraform-providers/terraform-provider-github/issues/469

### CODE REVIEW

File: examples/repository_collaborator/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Repository Collaborator
+
+This provides a template for managing [repository collaborators](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/inviting-collaborators-to-a-personal-repository).
+
+This example will also create a repository in the specified `owner` organization or a personal account. See https://www.terraform.io/docs/providers/github/index.html for details on configuring [`providers.tf`](./providers.tf) accordingly.
```
Comment (anGie44) [inline examples/repository_collaborator/README.md:5]: similar comment made in #478 -- we might want to remove reference to repo's made in a personal account for now?

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# GitHub Provider Examples
+
+This directory contains a set of examples of using various GitHub services with
+Terraform. The examples each have their own README containing more details
+on what the example does.
+
+To run any example, clone the repository and run `terraform apply` within
```
Comment (anGie44) [inline examples/README.md:7]: could we also specify the `init` and `plan` commands in the example? might be useful to add the complete flow for newer TF practitioners..hmm and possibly note to set variable definitions in a `.tfvars` file and pass as an arg to `apply` as one option

File: examples/repository_collaborator/README.md (hunk 1 above)
Comment (jcudit) [reply examples/repository_collaborator/README.md:5]:
```suggestion
This example will also create a repository in the specified `owner` organization. See https://www.terraform.io/docs/providers/github/index.html for details on configuring [`providers.tf`](./providers.tf) accordingly.
```

File: examples/README.md (hunk 2 above)
Comment (jcudit) [reply examples/README.md:7]:
```suggestion
To run any example, clone the repository and run `terraform init`, `terraform plan`, and `terraform apply` within
```

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+the example's own directory.
+
+For example:
+
+```
+$ git clone https://github.com/terraform-providers/terraform-provider-github
+$ cd terraform-provider-github/examples/repository_collaborator
+$ terraform apply
```
Comment (jcudit) [inline examples/README.md:15]:
```suggestion
$ terraform init
$ terraform plan
$ terraform apply
```

**Pull request merged**